tests/module_api/test_api.py

```diff
@@ -532,6 +534,38 @@ def test_update_membership(self):
         self.assertEqual(res["displayname"], "simone")
         self.assertIsNone(res["avatar_url"])
 
+        # Check that no remote join attempts have occurred thus far.
+        self.assertFalse(mocked_remote_join.called)
+
+        # Necessary to fake a remote join.
+        fake_stream_id = 1
+        if isinstance(mocked_remote_join, MagicMock):
+            # AyncMock is not supported in this Python version.
+            mocked_remote_join.return_value = make_awaitable(
+                ("fake-event-id", fake_stream_id)
+            )
+        else:
+            mocked_remote_join.return_value = "fake-event-id", fake_stream_id
```

Comment:
It would probably be more idiomatic to define the mock this way:

```python
mocked_remote_join = Mock(return_value=make_awaitable(("fake-event-id", fake_stream_id))
self.hs.get_room_member_handler()._remote_join = mocked_remote_join
```

Which would avoid having to patch it like that.

The second like of my proposal will likely anger mypy but that's something we usually happily silence with `# type: ignore[assignment]`.

Reply:
Done: a5721fa